oneoffs/distillation.py

```diff
@@ -0,0 +1,132 @@
+
+flags.DEFINE_integer("batch_size", 100, "batch_size for rotating.")
+flags.DEFINE_string("in_path", None, "tfrecord.zz to distill.")
+flags.DEFINE_string("out_path", None, "Records are writen to this file.")
+flags.DEFINE_string('model', 'saved_models/000721-eagle', 'Minigo Model')
+
+FLAGS = flags.FLAGS
+OPTS = tf.python_io.TFRecordOptions(tf.python_io.TFRecordCompressionType.ZLIB)
+
+def test():
+    records = list(tqdm(tf.python_io.tf_record_iterator(FLAGS.out_path, OPTS)))
+    print()
+    print("Values:")
+    for i, r in enumerate(records[:10]):
+        e = tf.train.Example.FromString(r)
+        print ("\t", i, e.features.feature['outcome'].float_list.value)
+
+
+def main(unusedargv):
+    in_path = FLAGS.in_path
+    out_path = FLAGS.out_path
+
+    assert tf.gfile.Exists(in_path)
+    #assert tf.gfile.Exists(os.path.dirname(out_path))
+
+    policy_err = []
+    value_err = []
+
+    with tf.python_io.TFRecordWriter(out_path, OPTS) as writer:
+        #for i, r in enumerate(tqdm(tf.python_io.tf_record_iterator(in_path, OPTS))):
+        #    writer.write(r)
+        #    if i == 2 ** 12 - 1:
+        #        break
+
+        ds_iter = preprocessing.get_input_tensors(
+            FLAGS.batch_size,
+            [in_path],
+            shuffle_examples=False,
+            random_rotation=False,
+            filter_amount=1.0)
+
+        with tf.Session() as sess:
+            features, labels = ds_iter
+            p_in = labels['pi_tensor']
+            v_in = labels['value_tensor']
+
+            p_out, v_out, logits = dual_net.model_inference_fn(
+                features, False, FLAGS.flag_values_dict())
+            tf.train.Saver().restore(sess, FLAGS.model)
+
+            for _ in tqdm(itertools.count(1)):
+                try:
+                    #x = sess.run(ds_iter)
+                    p_err = tf.nn.softmax_cross_entropy_with_logits_v2(
+                        logits=logits,
+                        labels=tf.stop_gradient(p_in))
+                    v_err = tf.square(v_out - v_in)
+
+                    x, pi, val, pi_err, val_err = sess.run(
+                        [features, p_out, v_out, p_err, v_err])
+
+                    # Undo cast in batch_parse_tf_example.
+                    x = np.int8(x)
+                    for i, pi, val in zip(range(FLAGS.batch_size), pi, val):
+                        record_out = preprocessing.make_tf_example(x[i], pi, val)
+                        serialized = record_out.SerializeToString()
+                        writer.write(serialized)
+
+                    policy_err.extend(pi_err)
+                    value_err.extend(val_err)
+
+                except tf.errors.OutOfRangeError:
+                    print()
+                    print("Breaking OutOfRangeError")
+                    break
+
+    print("Counts", len(policy_err), len(value_err))
+    test()
+
+    plt.subplot(121)
+    n, bins, patches = plt.hist(policy_err, 40)
+    plt.title('Policy Error histogram')
```

Comment:
I'd also be interested in seeing histograms for the following:
- entropy of the output distribution (`output_probs = tf.exp(logits); entropy = tf.reduce_sum(tf.log(output_probs) * output_probs)`

Reply:
oh actually I think it can also be computed as `entropy = tf.nn.softmax_cross_entropy_with_logits_v2(logits=logits, labels=logits)`